spopt/locate/frlm.py

```diff
@@ -0,0 +1,2089 @@
+        if objective == 'flow':
+            self.objective_value = final_evaluation['objective_value']
+        elif objective == 'vmt':
+            self.objective_value = sum(
+                final_evaluation['flow_coverage'][od_pair]['covered_proportion'] * 
+                self.flows[od_pair] * 
+                path_distances[od_pair]
+                for od_pair in final_evaluation['flow_coverage']
+            )
+        
+        self.solution_time = time.time() - start_time
+        self.status = "Heuristic"
+
+        self.solver_stats = {
+            'total_candidates_evaluated': sum(len(iter_info['candidates_evaluated']) for iter_info in iterations),
+            'average_candidates_per_iteration': np.mean([len(iter_info['candidates_evaluated']) for iter_info in iterations]),
+            'convergence_pattern': [iter_info['objective_value'] for iter_info in iterations],
+            'marginal_benefits': [iter_info['marginal_benefit'] for iter_info in iterations]
+        }
+        
+        print(f"\nGreedy Capacitated FRLM Solution:")
+        print(f"Objective value ({objective}): {self.objective_value:.2f}")
+        print(f"Total flow covered: {final_evaluation['covered_flow']:.2f} / {final_evaluation['total_flow']:.2f} ({final_evaluation['coverage_percentage']:.2%})")
+        print(f"Modules located at {len(self.selected_facilities)} locations:")
+        for k, count in self.selected_facilities.items():
+            print(f"  - Location {k}: {count} module(s)")
+        print(f"Solution time: {self.solution_time:.2f} seconds")
+        
+        result = final_evaluation.copy()
+        result.update({
+            'objective_value': self.objective_value,
+            'solution_time': self.solution_time,
+            'status': self.status,
+            'objective_type': objective
+        })
+        
+        if objective == 'vmt':
+            total_vmt = sum(self.flows[od_pair] * path_distances[od_pair] for od_pair in self.flows)
+            covered_vmt = self.objective_value
+            vmt_coverage_percentage = covered_vmt / total_vmt if total_vmt > 0 else 0
+            
+            result.update({
+                'total_vmt': total_vmt,
+                'covered_vmt': covered_vmt,
+                'vmt_coverage_percentage': vmt_coverage_percentage
+            })
+        
+        return result
+    
+    def _solve_pulp(self, 
+                   solver_instance: Optional[Any] = None,
+                   max_time_seconds: Optional[int] = 600,
+                   objective: str = "flow",
+                   weight_method: str = "double",
+                   **kwargs) -> Dict:
+
+        start_time = time.time()
+        self.solver_type = "pulp"
+        
+        if self.capacitated:
+            return self._solve_capacitated_pulp(solver_instance, max_time_seconds, objective, **kwargs)
+        else:
+            return self._solve_basic_pulp(solver_instance, max_time_seconds, weight_method, **kwargs)
```

Comment:
Again, these should be one method with one `if` statement depending on `self.capacity is None`.